core/tests/revert-test/tests/utils.ts

```diff
@@ -100,3 +108,389 @@ export function exec(command: string, options: ProcessEnvOptions) {
+        const depositHandle = await this.tester.syncWallet.deposit({
+            token: this.tester.isETHBasedChain ? zksync.utils.LEGACY_ETH_ADDRESS : this.tester.baseTokenAddress,
+            amount,
+            to,
+            approveBaseERC20: true,
+            approveERC20: true
+        });
+
+        let depositBatchNumber;
+        while (!(depositBatchNumber = (await depositHandle.wait()).l1BatchNumber)) {
+            console.log('Deposit is not included in L1 batch; sleeping');
+            await utils.sleep(1);
+        }
+        console.log(`Deposit was included into L1 batch ${depositBatchNumber}`);
+        expect(depositBatchNumber).to.be.greaterThan(initialL1BatchNumber);
+        return depositBatchNumber;
+    }
+}
+
+export class NodeSpawner {
+    public constructor(
+        private readonly pathToHome: string,
+        private readonly logs: fs.FileHandle,
+        private readonly fileConfig: FileConfig,
+        private readonly options: MainNodeSpawnOptions,
+        private readonly env?: ProcessEnvOptions['env']
+    ) {}
+
+    public async spawnMainNode(enableExecute: boolean): Promise<Node<NodeType.MAIN>> {
+        const env = this.env ?? process.env;
+        env.ETH_SENDER_SENDER_AGGREGATED_BLOCK_EXECUTE_DEADLINE = enableExecute ? '1' : '10000';
+        // Set full mode for the Merkle tree as it is required to get blocks committed.
+        env.DATABASE_MERKLE_TREE_MODE = 'full';
+
+        const { fileConfig, pathToHome, options, logs } = this;
+
+        if (fileConfig.loadFromFile) {
+            replaceAggregatedBlockExecuteDeadline(pathToHome, fileConfig, enableExecute ? 1 : 10000);
+        }
+
+        let components = 'api,tree,eth,state_keeper,commitment_generator,da_dispatcher,vm_runner_protective_reads';
+        if (options.enableConsensus) {
+            components += ',consensus';
+        }
+        if (options.baseTokenAddress != zksync.utils.LEGACY_ETH_ADDRESS) {
+            components += ',base_token_ratio_persister';
+        }
+        let proc = runServerInBackground({
+            components: [components],
+            stdio: ['ignore', logs, logs],
+            cwd: pathToHome,
+            env: env,
+            useZkInception: fileConfig.loadFromFile,
+            chain: fileConfig.chain
+        });
+
+        // Wait until the main node starts responding.
+        const tester = await Tester.init(
+            options.ethClientWeb3Url,
+            options.apiWeb3JsonRpcHttpUrl,
+            options.baseTokenAddress
+        );
+        await waitForNodeToStart(tester, proc, options.apiWeb3JsonRpcHttpUrl);
+        return new Node(tester, proc, NodeType.MAIN);
+    }
+
+    public async spawnExtNode(): Promise<Node<NodeType.EXT>> {
+        const env = this.env ?? process.env;
+        const { pathToHome, fileConfig, logs, options } = this;
+
+        let args = []; // FIXME: unused
+        if (options.enableConsensus) {
+            args.push('--enable-consensus');
+        }
```

Comment:
Is this intentional that `args` aren't passed anywhere? [Here's this code in `main`](https://github.com/matter-labs/zksync-era/blob/b8d4424c0be72a22df0c2a828785442822825b21/core/tests/revert-test/tests/revert-and-restart-en.test.ts#L264-L268).

Reply:
I think it's not we have to propagate it

Reply:
Will do so separately then to not block merging this PR.